lib/message/parser/tags.ts

```diff
@@ -27,12 +27,13 @@ export function parseTags(tagsSrc: string | undefined): IRCMessageTags {
   }
 
   for (const tagSrc of tagsSrc.split(";")) {
-    let key: string;
-    let valueSrc: string | undefined;
+    const key = tagSrc.split("=").shift() as string;
 
-    // eslint is bugged on this in the current version
-    // eslint-disable-next-line prefer-const
-    [key, valueSrc] = tagSrc.split("=", 2);
+    let valueSrc;
+    const indexOfFirstSeparator = tagSrc.indexOf("=");
+    if (indexOfFirstSeparator !== -1) {
+      valueSrc = tagSrc.slice(indexOfFirstSeparator + 1);
+    }
 
     tags[key.toLowerCase()] = decodeValue(valueSrc);
```

Comment:
```suggestion
    const indexOfFirstSeparator: number = tagSrc.indexOf("=");

    if (indexOfFirstSeparator === -1) continue;

    const key: string = tagSrc.slice(0, indexOfFirstSeparator),
      valueSrc: string = tagSrc.slice(indexOfFirstSeparator + 1);

    tags[key.toLowerCase()] = decodeValue(valueSrc);
```
I'm not positive if this works, but from just reading the code, this is what came to mind.

(Probably for a future PR) Imo, moving to a traditional for loop, and slicing indices here instead of splitting at all, seems like the move, and would reduce a lot of wasted allocation and looping, not sure why it was written like this in the first place.

Reply:
Thank you, please consider creating issues for those. You're free to take up implementing them.